### ISSUE TITLE
Support invite code

### ISSUE THREAD

**Issue by mooreds** [FusionAuth/fusionauth-issues#1335]
### Problem
I want the ability to require an invite code for user registration.

### Solution
Have a way to set up invite codes. These could be tied to a user or an application. Would be nice to have a number associated with them (so that, say, 10 people could use invite code 'dan' and 20 people could use invite code 'foo', but no more). Have a setting per application on whether to require an invite code. 

When a user registers with the hosted login pages, require them to provide a valid invite code. If they don't have one or it has been fully used, provide the appropriate error message.

### Alternatives/workarounds
You can code this all up yourself using the Login API.

### Community guidelines
All issues filed in this repository must abide by the [FusionAuth community guidelines](https://fusionauth.io/community/forum/topic/1000/code-of-conduct).

### How to vote
Please give us a thumbs up or thumbs down as a reaction to help us prioritize this feature. Feel free to comment if you have a particular need or comment on how this feature should work.

**Comment by jobannon** [comment]
Seen on https://fusionauth.io/community/forum/topic/1351/send-additional-params-to-identity-providers-link/4?_=1636749387444

**Comment by mooreds** [comment]
Here's another option: https://fusionauth.io/community/forum/topic/2650/how-to-implement-user-invitation

**Comment by th30d0r** [comment]
I mean this is workaround not a feature, also it does require quite a few steps, for something that is quite a straight forward practice

**Comment by mooreds** [comment]
@th30d0r You are correct. I can't see whether or not you upvoted this issue, but if not, please do so.
